### PR TITLE
[fbcnms-alarms]: Use material-table

### PR DESCRIPTION
**Pull request by HannaFar**
use material-table 

### alerts table
before
Capture d’écran 2021-07-16 à 12 38 35

after
Capture d’écran 2021-07-16 à 12 32 50

### rules table
before
Capture d’écran 2021-07-16 à 12 38 47

after
Capture d’écran 2021-07-16 à 12 33 43

### teams table
before
Capture d’écran 2021-07-16 à 12 38 57

after
Capture d’écran 2021-07-16 à 12 33 32

### routes table
before
Capture d’écran 2021-07-16 à 12 53 17

after
Capture d’écran 2021-07-16 à 12 55 10

### suppressions table
before
Capture d’écran 2021-07-16 à 12 47 32

after
Capture d’écran 2021-07-16 à 12 55 21

### CODE REVIEW

File: fbcnms-packages/fbcnms-alarms/components/AlertRules.js [hunk 1]
```diff
@@ -70,57 +66,46 @@ export default function AlertRules<TRuleUnion>() {
...
 
-  const columnStruct = React.useMemo<
-    Array<ColumnData<GenericRule<TRuleUnion>>>,
-  >(
+  const columnStruct1 = React.useMemo(
     () => [
       {
         title: 'name',
-        getValue: x => x.name,
+        field: 'name',
```
Comment (andreilee) [inline fbcnms-packages/fbcnms-alarms/components/AlertRules.js:78]: Capitalize the column titles

File: fbcnms-packages/fbcnms-alarms/components/alertmanager/Suppressions.js [hunk 2]
```diff
@@ -67,22 +66,28 @@ export default function Suppressions() {
...
-          setCurrentRow(alert);
-        }}
+        onRowClick={row => setCurrentRow(row)}
         columnStruct={[
-          {title: 'name', getValue: row => row.comment || ''},
-          {title: 'active', getValue: row => row.status?.state ?? ''},
-          {title: 'created by', getValue: row => row.createdBy},
+          {title: 'name', field: 'comment'},
+          {title: 'active', field: 'status.state'},
+          {title: 'created by', field: 'createdBy'},
```
Comment (andreilee) [inline fbcnms-packages/fbcnms-alarms/components/alertmanager/Suppressions.js:73]: Capitalize the column titles here as well

File: fbcnms-packages/fbcnms-alarms/components/alertmanager/Suppressions.js [hunk 3]
```diff
@@ -67,22 +66,28 @@ export default function Suppressions() {
...
                 ? row.matchers.map(matcher => toLabels(matcher))
-                : [],
-            render: 'multipleGroups',
+                : [];
+              return <MultiGroupsCell value={value} />;
+            },
+          },
+        ]}
+        tableData={silencesList || []}
+        data-testid="suppresions"
```
Comment (andreilee) [inline fbcnms-packages/fbcnms-alarms/components/alertmanager/Suppressions.js:86]: typo here

File: fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js [hunk 4]
```diff
@@ -9,169 +9,107 @@
...
-    }
-  }
-}
+type ActionTableOptions = {
+  actionsColumnIndex: number,
+  pageSize?: number,
+  pageSizeOptions: Array<number>,
+  rowStyle?: {},
+  headerStyle?: {},
+};
```
Comment (andreilee) [inline fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js:76]: Add comments to describe how to use these options

**Review by andreilee — COMMENTED** [review]
Please make sure to add before/after screenshots before publishing.

Considering the frequent re-use of `SimpleTable`, I think there needs to be some more extensive comments/documentation on usage, as well as descriptions of various arguments/options.

File: fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js [hunk 5]
```diff
@@ -183,205 +121,166 @@ const renderLabelValue = (labelValue: LabelVal) => {
...
+  value: TValue,
+};
 type LabelVal = string | number | boolean;
 type Labels = {[string]: LabelVal};
-function LabelsCell({
+export function LabelsCell({
   value,
-  classes,
-  columnIdx,
-  hideFields,
```
Comment (HannaFar) [inline fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js:192]: This wasn't used so I removed it

File: fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js [hunk 6]
```diff
@@ -183,205 +121,166 @@ const renderLabelValue = (labelValue: LabelVal) => {
...
-
-function SeverityCell({value}: CellProps<string>) {
-  return (
-    <BodyTableCell>
-      <SeverityIndicator severity={value} />
-    </BodyTableCell>
-  );
-}
-
-function ChipCell({value, classes}: CellProps<string>) {
```
Comment (HannaFar) [inline fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js:249]: This wasn't used

File: fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js (hunk 4 above)
Comment (HannaFar) [reply fbcnms-packages/fbcnms-alarms/components/table/SimpleTable.js]: I removed rowstyle and headerstyle, we are not using it

**Review by andreilee — CHANGES_REQUESTED** [review]
Could you rebase and bump the version number of `@fbcnms/alarms` to 0.4.0?

**Review by andreilee — APPROVED** [review]

**Pull request merged**